### ISSUE TITLE
istio-clean-iptables does not clean up extra kube-virt-interfaces

### ISSUE THREAD

**Issue by dwj300** [istio/istio#48368]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

We are trying to use kube-virt-interfaces to capture traffic from inside docker bridge networks on our EC2 hosts. Unfortunately, istio-iptables is failing after running `pilot-agent istio-clean-iptables` because it is trying to create a chain that already exists. Here is a repro:
```
$ pilot-agent istio-iptables \
  --envoy-port 15001 \
  --inbound-capture-port 15006 \
  --istio-inbound-interception-mode 'REDIRECT' \
  --istio-service-cidr '100.124.0.0/15' \
  --proxy-gid 1337 \
  --proxy-uid 1337 \
  --kube-virt-interfaces docker0
...
$ pilot-agent istio-clean-iptables  
...
$ pilot-agent istio-iptables \
  --envoy-port 15001 \
  --inbound-capture-port 15006 \
  --istio-inbound-interception-mode 'REDIRECT' \
  --istio-service-cidr '100.124.0.0/15' \
  --proxy-gid 1337 \
  --proxy-uid 1337 \
  --kube-virt-interfaces docker0
  ...
  2023-12-14T18:54:50.064433Z     info    Writing following contents to rules file: /tmp/iptables-rules-1702580090064382179.txt3999944652
* nat
-N ISTIO_INBOUND
-N ISTIO_REDIRECT
-N ISTIO_IN_REDIRECT
-N ISTIO_OUTPUT
-I PREROUTING 1 -i docker0 -j RETURN
-A ISTIO_INBOUND -p tcp --dport 15008 -j RETURN
-A ISTIO_REDIRECT -p tcp -j REDIRECT --to-ports 15001
-A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-ports 15006
-A OUTPUT -p tcp -j ISTIO_OUTPUT
-A ISTIO_OUTPUT -o lo -s 127.0.0.6/32 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --uid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --gid-owner 1337 -j ISTIO_IN_REDIRECT
-A ISTIO_OUTPUT -o lo -m owner ! --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -m owner --gid-owner 1337 -j RETURN
-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
-I PREROUTING 1 -i docker0 -d 100.124.0.0/15 -j ISTIO_REDIRECT
-A ISTIO_OUTPUT -d 100.124.0.0/15 -j ISTIO_REDIRECT
-A ISTIO_OUTPUT -j RETURN
COMMIT
2023-12-14T18:54:50.064467Z     info    Running command: iptables-restore --noflush /tmp/iptables-rules-1702580090064382179.txt3999944652
2023-12-14T18:54:50.065202Z     error   Command error output: xtables other problem: line 3 failed
2023-12-14T18:54:50.065229Z     error   Failed to execute: iptables-restore --noflush /tmp/iptables-rules-1702580090064382179.txt3999944652, exit status 1
  ```

To fix, we need to run
```
$ sudo iptables -t nat -D PREROUTING 1
$ sudo iptables -t nat -X ISTIO_REDIRECT
```

### Version

```Text
istio 1.17.4
```

### Additional Information

Here are the lingering rules after the cleanup:

```
$ sudo iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination
ISTIO_REDIRECT  all  --  anywhere             100.124.0.0/15
RETURN     all  --  anywhere             anywhere
DOCKER     all  --  anywhere             anywhere             ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  anywhere            !ip-127-0-0-0.ec2.internal/8  ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
MASQUERADE  all  --  ip-172-17-0-0.ec2.internal/24  anywhere

Chain DOCKER (2 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere

Chain ISTIO_REDIRECT (1 references)
target     prot opt source               destination
```

**Comment by dwj300** [comment]
not stale

**Comment by howardjohn** [comment]
cc @leosarra this might have been fixed by the work you did recently

**Comment by leosarra** [comment]
> cc @leosarra this might have been fixed by the work you did recently

I agree, this seems like something that should already be fixed by the new cleanup methodology in the `--cleanup-only` arg.
This option will be available starting with Istio 1.24.x.
However, I think we should still ship a fix for older releases (1.23/1.22), as they will not receive the new `--cleanup-only` arg.

I will try to work on a fix for the legacy istio-clean-iptables at the end of this week or the start of next one.
